### PR TITLE
Changes to the User Guide's Entry Points page

### PR DESCRIPTION
**Pull request by codeandfire**
## Summary of changes

Console scripts section:
 - Added `pyproject.toml` snippet
 - Modified `setup.py` snippet (corrected indentation and removed unnecessary name, version, etc. fields)
 - Changed the example to use src-layout
 - Some changes in the wording etc.

Added GUI Scripts section:
 - Included `setup.cfg` / `setup.py` / `pyproject.toml` snippets
 - Added an example Hello world function that can be invoked through a GUI script

Other changes:
 - Added note on distinction between Console and GUI scripts
 - Added note on how installers like Pip create wrapper scripts around invoked functions
 - Included few lines stating that invoked functions should not accept any arguments and parse user input within the function body

_Have not finished this yet. Will update here with further changes._

### Pull Request Checklist
- [ ] News fragment added in [`changelog.d/`].
  _(See [documentation][PR docs] for details)_

[`changelog.d/`]: https://github.com/pypa/setuptools/tree/master/changelog.d
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:104]:
```suggestion
the function to parse user input given via :obj:`sys.argv`.
```

Maybe mention `sys.argv` directly so users are aware that they can also access it?

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:130]: Cool example!

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:163]: The note may be obvious, but maybe it is helpful for beginners? Or should we instead just add the dependency requirements directly in the tabs above?

```suggestion

.. note::
   To be able to import `PySimpleGUI`, you need to add `pysimplegui` to your package dependencies.
   See :doc:`/userguide/dependency_management` for more information.
```

**Review by abravalheri — APPROVED** [review]
Thank you very much for all the improvements @codeandfire!

File: docs/userguide/entry_point.rst
Comment (codeandfire) [reply docs/userguide/entry_point.rst:162]: Yeah, I thought about this too. Your call on this. I think both ways will be alright.

File: docs/userguide/entry_point.rst
Comment (codeandfire) [reply docs/userguide/entry_point.rst:129]: Thanks!

File: docs/userguide/entry_point.rst
Comment (abravalheri) [reply docs/userguide/entry_point.rst:162]: Let's add the note, *just-in-case* ™️

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:164]:
```suggestion
   To be able to import ``PySimpleGUI``, you need to add ``pysimplegui`` to your package dependencies.
```

Yeah, I forgot the double backticks... This constant change between Markdown (used by Github) and RST used by sphinx...

**Comment by codeandfire** [comment]
Sure @abravalheri, I just have some changes in mind for the second part of the page involving the plugin-related entry points, and I think I will finish them in a day. After that can we merge this commit?

**Comment by codeandfire** [comment]
Hi @abravalheri, a bit late but I've finished my work on the Entry Points page. I have written a lot of new material, please let me know your feedback!

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:269]: I think we might be able to use intersphinx here:

```suggestion
:mod:`importlib.metadata` module,
```

File: docs/userguide/entry_point.rst
Comment (abravalheri) [inline docs/userguide/entry_point.rst:469]:
```suggestion
:mod:`importlib.metadata` module,
```

**Review by abravalheri — APPROVED** [review]
Thank you very much @codeandfire.
The changes in the documentation are superb, I am very glad to have you as a contributor.

File: docs/userguide/entry_point.rst
Comment (abravalheri) [reply docs/userguide/entry_point.rst]: Since these intersphinx changes are very small and don't modify the meaning of the PR, I went ahead and already merged them.

**Comment by codeandfire** [comment]
Thank you @abravalheri for your kind words. It's a pleasure to contribute to Setuptools, especially when you as a maintainer are so helpful and responsive!

**Pull request merged**
